### PR TITLE
Restore custom `distutils` handling for resolving paths to submodules.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
I understand @DanielNoord wants to bisect before evaluating a path forward. But this test case fails without these changes, so it could be a sufficient way forward.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

Refs https://github.com/PyCQA/pylint/issues/5645

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -406,3 +406,29 @@ jobs:
         run: |
           . venv/bin/activate
           pytest tests/
+
+  extra-special-test-case:
+    name: Test case that requires specific virtualenv
```
Comment (jacobtylerwalls) [inline .github/workflows/ci.yaml:411]: Obviously we're not going to merge it like this.

**Comment by DanielNoord** [comment]
I wonder how we can fully test this..
This does seem like it would work though.

@cdce8p What do you think? Try this and see if the problems go away?

**Comment by jacobtylerwalls** [comment]
> I wonder how we can fully test this..

I mean, we _could_ add the CI step in this PR if we really want to have an extra workflow that only runs one corner case!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jacobtylerwalls) [reply .github/workflows/ci.yaml]: I improved this CI step to give us the option of merging it.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you @jacobtylerwalls. We just need to remove the temporary CI job before merging, right ?

**Comment by DanielNoord** [comment]
Do we though? Won't the CI job be a nice test to see this actually works?

**Comment by jacobtylerwalls** [comment]
Fine with me. I added caching to make it more efficient.

**Review by cdce8p — APPROVED** [review]
The change looks like it should work! I'm just not sure that we should run the CI job for every time. It's excellent to have but maybe better as part of an optional CI suite?

What do you guys think about moving it to a separate workflow file and using the `workflow_dispatch` event to trigger it whenever we need to? For that case, it would make sense to add a small comment in the code mentioning the workflow and that it can be run manually. Lastly, the caching could probably be removed. Although slightly more efficient, it can also create unexpected issues. If only run on request, there likely won't be any cache which can be restored to begin with.

An alternative would be to add `if: false` to the job. It would also be trivial to change it to `true` whenever we need to work on it (which itself shouldn't be all too often).

After writing this, I'm just slightly leaning towards the latter option.

**Comment by jacobtylerwalls** [comment]
I'm thinking along the lines of the former option: what about creating a "release testing" CI workflow with `on: workflow_dispatch`, and then adding a reminder to the release.md checklist to remember to run it before releases?

**Comment by cdce8p** [comment]
> The issue is then that we can tag buggy commit and have nothing released.

The idea was to have run automatically for every release. I still wanted to keep the manual trigger.
You do have a point though that deleting a tag is quite some work. Let's do just the `workflow_dispatch` trigger then with a note in `release.md` as suggested.

File: .github/workflows/release-tests.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: Release tests
+
+on: workflow_dispatch
+
+env:
+  DEFAULT_PYTHON: 3.8
+
+jobs:
+  virtualenv-15-windows-test:
+    name: Regression test for virtualenv==15.1.0 on Windows
```
Comment (cdce8p) [inline .github/workflows/release-tests.yml:10]: Last two 🙂

I believe it's possible to add comments in workflow files too. It would be great if we could link to the PR here just to have a reference later why it was created.

File: astroid/interpreter/_import/spec.py [hunk 3]
```diff
@@ -160,6 +161,20 @@ def contribute_to_path(self, spec, processed):
                 for p in sys.path
                 if os.path.isdir(os.path.join(p, *processed))
             ]
+        elif spec.name == "distutils":
+            # virtualenv below 20.0 patches distutils in an unexpected way
+            # so we just find the location of distutils that will be
+            # imported to avoid spurious import-error messages
```
Comment (cdce8p) [inline astroid/interpreter/_import/spec.py:167]: A PR (or issue) link could also be helpful here.

**Comment by cdce8p** [comment]
Thanks @jacobtylerwalls 🐬

**Comment by DanielNoord** [comment]
@ everybody, do we want to test this as well in `pylint`? I don't think so right? (Seeing the difficulty of doing so in `astroid`). If not we can also close the `pylint` issue I think.

**Comment by cdce8p** [comment]
> @ everybody, do we want to test this as well in `pylint`? I don't think so right? (Seeing the difficulty of doing so in `astroid`). If not we can also close the `pylint` issue I think.

Yeah, let's close the `pylint` issue. The test here should be enough.
